tools/tools.go

```diff
@@ -314,6 +318,71 @@ func EnvmanRun(envstorePth, workDirPth string, cmd []string) (int, error) {
 	return command.RunCommandInDirAndReturnExitCode(workDirPth, "envman", args...)
 }
 
+// EnvmanRunWithTimeout ...
+func EnvmanRunWithTimeout(envstorePth, workDirPth string, cmdArgs []string, timeout time.Duration) (int, error) {
+	logLevel := log.GetLevel().String()
+	args := []string{"--loglevel", logLevel, "--path", envstorePth, "run"}
+	args = append(args, cmdArgs...)
+
+	command := command.NewWithStandardOuts("envman", args...).SetStdin(os.Stdin).SetDir(workDirPth)
+
+	// create a new process group for our process and its child processes
+	cmd := command.GetCmd()
+	cmd.SysProcAttr = &syscall.SysProcAttr{Setpgid: true}
+
+	if err := cmd.Start(); err != nil {
+		return 1, err
```

Comment:
this is an internal error, I'd suggest to use exit code < 0, e.g. `-1` as this is not the exit code of the command